tests/Test.ADAL.NET.Integration/SovereignCloudsTests.cs

```diff
@@ -60,16 +97,90 @@ public void Initialize()
+            Assert.AreEqual(1, authenticationContext.TokenCache.tokenCacheDictionary.Count);
+            Assert.AreEqual(sovereignTenantSpecificAuthority,
+                authenticationContext.TokenCache.tokenCacheDictionary.Keys.FirstOrDefault().Authority);
+
+            // all mocks are consumed
+            Assert.AreEqual(0, HttpMessageHandlerFactory.MockHandlersCount());
+        }
+
+        [TestMethod]
+        [Description("Instance discovery call is made because authority was not already in the instance cache")]
+        public async Task AuthorityNotInInstanceCache_InstanceDiscoverCallMadeTestAsync()
+        {
+            HttpMessageHandlerFactory.InitializeMockProvider();
+
+            string content = @"{
+                            ""tenant_discovery_endpoint"":""https://login.microsoftonline.com/tenant/.well-known/openid-configuration"",
+                            ""api-version"":""1.1"",
+                            ""metadata"":[{
+                                ""preferred_network"":""login.microsoftonline.com"",
+                                ""preferred_cache"":""login.windows.net"",
+                                ""aliases"":[
+                                    ""login.microsoftonline.com"",
+                                    ""login.windows.net"",
+                                    ""login.microsoft.com"",
+                                    ""sts.windows.net""]}]}";
+
+            // creating AuthenticationContext with common Authority
+            var authenticationContext =
+                new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, false, new TokenCache());
+
+            // mock value for authentication returnedUriInput, with cloud_instance_name claim
+            var authReturnedUriInputMock = TestConstants.DefaultRedirectUri + "?code=some-code" + "&" +
+                                           TokenResponseClaim.CloudInstanceHost + "=" + sovereignAuthorityHost;
+
+            MockHelpers.ConfigureMockWebUI(
+                new AuthorizationResult(AuthorizationStatus.Success, authReturnedUriInputMock),
+                // validate that authorizationUri passed to WebUi contains instance_aware query parameter
+                new Dictionary<string, string> { { "instance_aware", "true" } });
+
+            HttpMessageHandlerFactory.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.GetDiscoveryEndpoint(TestConstants.DefaultAuthorityCommonTenant), content));
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Get,
+                ResponseMessage =
+                    MockHelpers.CreateSuccessTokenResponseMessage(TestConstants.DefaultUniqueId,
+                    TestConstants.DefaultDisplayableId, TestConstants.DefaultResource)
+            });
+
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
```

Comment:
why is this mock needed?

Reply:
I removed it...needed to move a few things around.